src/app.ts

```diff
@@ -50,104 +49,42 @@ const extractPullRequestInformation = (payload: PullRequestEvent) => {
-      return validatedTemplate.errors.concat({
-        message: `Version could not be found at ${error.owner}/${error.repo}:${error.version}`,
-        line: undefined,
-      })
-    } else {
-      throw error
-    }
+const extractCheckRunInformation = (payload: CheckRunRerequestedEvent) => {
+  const {
+    check_run: { pull_requests, id },
+    repository,
+  } = payload
+
+  if (pull_requests.length != 1) return undefined
+  return {
+    number: pull_requests[0].number,
+    sha: pull_requests[0].head.sha,
+    checkId: id,
+    repository: {
+      owner: repository.owner.login,
+      repo: repository.name,
+      defaultBranch: repository.default_branch,
+    },
   }
 }
 
-const processPullRequest = async (payload: PullRequestEvent, context: Context<'pull_request'>) => {
-  const { number: prNumber, sha, repository } = extractPullRequestInformation(payload)
+const processCheckRerun = async (payload: CheckRunRerequestedEvent, context: Context<'check_run'>) => {
+  const extracted = extractCheckRunInformation(payload)
+  if (extracted === undefined) return
+  const { number: prNumber, sha, repository, checkId } = extracted
+
   const { log, octokit } = context
   const enrichedWithRepoLog = log.child({ owner: repository.owner, repository: repository.repo })
 
-  const { approvePullRequestChanges, getFilesChanged, requestPullRequestChanges } = git(enrichedWithRepoLog, octokit)
-
-  const { createCheckRun, resolveCheckRun } = checks(enrichedWithRepoLog, octokit)
-  const { determineConfigurationChanges } = configuration(enrichedWithRepoLog, octokit)
-  const conclusion = (errors: ValidationError[]) => (errors.length > 0 ? 'failure' : 'success')
-
-  log.info('Pull request event happened on #%d', prNumber)
-
-  const filesChanged = await getFilesChanged(repository, prNumber)
-  const configFile = filesChanged.find(filename => filename === configFileName)
-
-  if (!configFile) return
-
-  const checkInput = {
-    ...repository,
-    sha: sha,
-  }
-  const checkId = await createCheckRun({
-    ...repository,
-    sha: sha,
-  })
-
-  log.debug('Found repository configuration file: %s.', configFile)
-
-  const configurationChanges = await determineConfigurationChanges(configFileName, repository, sha)
-
-  const errors = await validateChanges(log, octokit, configurationChanges)
-
-  const onlyChangesConfiguration = filesChanged.length === 1 && filesChanged[0] === configFileName
-
-  if (errors.length > 0) {
-    const changeRequestId = await requestPullRequestChanges(repository, prNumber, checkId)
-    log.debug(`Requested changes for PR #%d in %s.`, prNumber, changeRequestId)
-  } else if (onlyChangesConfiguration) {
-    const approvedReviewId = await approvePullRequestChanges(repository, prNumber)
-    log.debug(`Approved PR #%d in %s.`, prNumber, approvedReviewId)
-  }
+  await processCheckRun(enrichedWithRepoLog, octokit, configFileName, prNumber, repository, sha, checkId)
```

Comment:
I think we should use the same pattern here for the functions that have been extracted as in other modules. That way we don't pass 7 parameters in a single function and can instead encapsulate the logger and Octokit like so:

```suggestion
//Outside of this function
import { validation } from './core-validation'
  //...
  const { processCheckRun } = validation(enrichedLogger, octokit)
  await processCheckRun(configFileName, prNumber, repository, sha, checkId)
```

I still think 5 parameters is too much (does the function do more than one thing or can we combine the data?), but that can be fixed later